src/main/java/de/craften/ui/swingmaterial/MaterialWindow.java

```diff
@@ -8,29 +8,216 @@
+        } else {
+            contentPane = super.getContentPane();
         }
+        getRootPane().putClientProperty("Window.shadow", Boolean.FALSE);
+        getRootPane().setWindowDecorationStyle(JRootPane.NONE);
+        addComponentListener(new ComponentAdapter() {
+            @Override
+            public void componentResized(ComponentEvent componentEvent) {
+                setShape(new RoundRectangle2D.Float(0, 0, getWidth(), getHeight(), 3, 3));
+            }
+        });
+        light();
     }
 
-    private class ShadowPane extends JPanel {
+    @Override
+    public Container getContentPane() {
+        return contentPane;
+    }
+
+    @Override
+    public void setContentPane(Container contentPane) {
+        this.contentPane = contentPane;
+    }
+    
+    /**
+     * Chainable method to quickly get a {@link MaterialWindow} with a dark
+     * background and white foreground. You can append it to your constructor
+     * like this: {@code MaterialWindow myWindow = new MaterialWindow.dark()} or
+     * just use it alone to switch the foreground/background colors.
+     * @return this window
+     */
+    public MaterialWindow dark() {
+        getContentPane().setBackground(MaterialColor.GREY_900);
+        getContentPane().setForeground(MaterialColor.WHITE);
+        return this;
+    }
+    
+    /**
+     * Chainable method to quickly get a {@link MaterialWindow} with a white
+     * background and black foreground. You can append it to your constructor
+     * like this: {@code MaterialWindow myWindow = new MaterialWindow.dark()} or
+     * just use it alone to switch the foreground/background colors.
+     * @return this window
+     */
+    public MaterialWindow light() {
+        getContentPane().setBackground(MaterialColor.WHITE);
+        getContentPane().setForeground(MaterialColor.GREY_900);
+        return this;
+    }
+    
+    /**
+     * Alternative method to {@link JFrame#setLocation(int, int)}. Use it to set
+     * a new location on screen for this window, this method takes into
+     * consideration the shadow offsets so you don't have to worry about it.
+     * @param x the x-coordinate of the new location's top-left corner in the desktop
+     * @param y the y-coordinate of the new location's top-left corner in the desktop
+     */
+    public void setWindowLocation(int x, int y) {
+        if (shadowPane == null) {
+            super.setLocation(x, y);
+        } else {
+            super.setLocation(x - MaterialShadow.OFFSET_LEFT, y - MaterialShadow.OFFSET_TOP);
+        }
+    }
+    
+    /**
+     * Alternative method to {@link JFrame#setLocation(java.awt.Point)}. Use it
+     * to set a new location on screen for this window, this method takes into
+     * consideration the shadow offsets so you don't have to worry about it.
+     * @param p the point defining the top-left corner of the new location in the desktop
+     */
+    public void setWindowLocation(Point p) {
```

Comment:
This should just call `this.setWindowLocation(p.x, p.y);` instead of re-implementing that logic.